core/src/test/scala/unit/kafka/server/DynamicConfigChangeTest.scala

```diff
@@ -309,19 +310,20 @@ class DynamicConfigChangeTest extends KafkaServerTestHarness {
     val metrics =
       "org.apache.kafka/client.producer.partition.queue.,org.apache.kafka/client.producer.partition.latency"
     val pushInterval = 30 * 1000 // 30 seconds
-    val pattern1 = CmClientInformation.CLIENT_SOFTWARE_NAME + " = Java       "
-    val pattern2 = CmClientInformation.CLIENT_SOFTWARE_VERSION + " =     11.*   "
+    val pattern1 = CLIENT_SOFTWARE_NAME + " = Java       "
+    val pattern2 = CLIENT_SOFTWARE_VERSION + " =     11.*   "
     val patternsList = List(pattern1, pattern2)
 
     val props = ClientMetricsTestUtils.getDefaultProperties()
 
     // ********  Test Create the new client subscription with multiple client matching patterns *********
     updateClientSubscription(configEntityName, props,
-                             () =>  ClientMetricsConfig.getClientSubscriptionGroup(configEntityName) != null)
-    val sgroup = ClientMetricsConfig.getClientSubscriptionGroup(configEntityName)
-    assertTrue(sgroup.getPushIntervalMs == pushInterval)
-    assertTrue(sgroup.getSubscribedMetrics.size == 2 && sgroup.getSubscribedMetrics.mkString(",").equals(metrics))
-    assertTrue(sgroup.getClientMatchingPatterns.size == patternsList.size)
+                             () =>  ClientMetricsConfig.getClientSubscriptionInfo(configEntityName) != null)
+    val cmSubscription = ClientMetricsConfig.getClientSubscriptionInfo(configEntityName)
+    assertTrue(cmSubscription.getPushIntervalMs == pushInterval)
```

Comment:
maybe assertEqual here? maybe also add failure assertion message there?

Reply:
I am just curious, is there any advantage to use assertEqual instead of assertTrue ?

Reply:
I think for using AssertEquals you get a more sensible message like: Expected A but should be B, instead of Expected true, but was false.  I think it provides better clarity when tests are failing.